### ISSUE TITLE
OpenCilk academic goals 2022-2023

### ISSUE THREAD

**Issue by behoppe** [OpenCilk/www.opencilk.org#192]
Thank you @jowens for following draft list of academic goals. I have created this issue as a place to discuss our progress. (These are the content-related goals, as I expect we'll manage the outreach-related goals on our CRM.)
1. (Published on website) List of faculty currently offering related courses, with appropriate links. (BEH: We will add these to https://www.opencilk.org/community/teach-performance/. The relevant issue is #9.)
1. (Privately available) Repository of teaching materials, with an agreement that they can be used within the “circle of trust of faculty” and not otherwise made public. (BEH: yes, and we'll create the private repo along lines of issue #42.) 
1. Populated low-traffic mailing list of interested faculty. (BEH: yes. IMO the key ingredients here are the IDs of interested faculty and the content that we'll distribute. We already have several low-traffic mailing lists and a history of not using them. When we get to the mailing list part of this, the relevant issue is #54.) 
1. (Published on website) “Model” / “sample” curricula, primarily at this point patterned exactly after curricula that have already taught. (BEH: great idea. I assume we'll include this with the top-most item?)

cc: @cleiserson @neboat

**Comment by behoppe** [comment]
Thanks @jowens for creating the Box folder for our privately available repository of teaching materials, which updates item #2 above. 

This raises a couple interrelated questions: 
* Do we have any need for the teaching-resources repository on GitHub?
* Do we still aim to use GitHub teams to administer community membership and content access?

**Comment by behoppe** [comment]
Hi @jowens @neboat @cleiserson @DorothyCurtis, now that community members are asking me to (securely) provide all class materials from 6.172 of 2020, I'm reconsidering if/how to continue using Box.com. I'm posting here because the top of this thread holds all the relevant context (including big open questions about how exactly we use GitHub teams).

TL;DR -- I recommend migrating all non-video content to our original teaching materials repository, so that access can be controlled with GitHub teams. I think our videos can be shared publicly, which means we might put them on a YouTube channel; or we can continue hosting them on Box.

More detail:

If we consider (1) hosting content securely for teachers, (2) hosting huge files (i.e., videos), and (3) managing access with the same system that controls our code, it seems that we have to use at least two platforms to do this -- GitHub and something.

Right now we are using only Box and ignoring GitHub access control. IMO that's a short-term solution, and we should do what we can to preserve our ability to use GitHub teams. (Otherwise our entire community access system needs to be re-imagined.)

By migrating non-video files back to GitHub, we make it possible to use GitHub teams for everything but the videos. I think we don't need any access control for any of our videos, so this works OK. We can continue to host videos on Box, or we might make a YouTube channel.

What do you think?

**Comment by jowens** [comment]
I'm comfortable with @behoppe's recommendation.

**Comment by behoppe** [comment]
@jowens thanks for leading our professor outreach. As we email-discussed, we've learned a lot and we're shifting gears. I wanted to log that here along with next steps that I think/hope you're on board to do:
* Write an article about "so you want to teach a course about performance engineering" so we can put it on the website and then point people to it, ask for feedback, and offer to talk to them.  (Thank you)
* Could you also write a summary of what you/we have learned from the interviews? I think article 1 is a distillation of what helps prospective teachers, and article 2 is a distillation that helps the OpenCilk team prioritize next steps. The rest of us have a sense of this, but I think we'll miss out if we don't hear your voice on this.

I am ready to help anytime with editing and publishing.
